### PR TITLE
Feature/parse reads for gender

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Adds modules to parse fastq files to estimate gender and avoid having to be wrong 50% of the time
- Will only be performed for DNA WGS samples
- Works for single/paired end and interleaved fastq files
- Triggers automatically when any samples do not have "male" or "female" gender set.
- Updated contig info accordingly
- Adds a entry to analysis config showing the estimated gender, but leaves original pedigree untouched
- Estimation takes less than 5 seconds
- Modifies use of found_[GENDER] keys internally

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
@moahaegglund The code is all but done! Now we just need the science behind these values of these parameters:
```
$MALE_THRESHOLD = Number of Y read count to assign male gender (Now at 25)
$BYTE_START_POS = Where to start the byte chunk stream in the fastq files (Now at 10,000)
$BYTE_STOP_POS  => The length of the byte chunk (Now at: $BYTE_START_POS * 100;)

File: lib/MIP/Main/Analyse.pm [hunk 1]
```diff
@@ -191,9 +191,8 @@ sub mip_analyse {
           load_yaml( { yaml_file => $active_parameter{config_file}, } );
 
         ## Remove previous analysis specific info not relevant for current run e.g. log file, which is read from pedigree or cmd
-        my @remove_keys = (
-            qw{ found_female found_male found_other found_other_count log_file dry_run_all }
-        );
+        my @remove_keys =
+          (qw{ found_female found_male found_other gender log_file dry_run_all });
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply lib/MIP/Main/Analyse.pm]: No, they should not. Will remove.

Comment (henrikstranneheim) [inline lib/MIP/Main/Analyse.pm]:
```suggestion
          qw{ found_female found_male found_other gender log_file dry_run_all };
```

File: lib/MIP/Parse/Gender.pm [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+    use File::Path qw{ remove_tree };
+
+    ## Constants
+    Readonly my $MAX_RANDOM_NUMBER => 10_000;
+
+    ## Generate a random integer between 0-10,000.
+    my $random_integer = int rand $MAX_RANDOM_NUMBER;
+
+    ## Temporary bash file for commands
+    my $bash_temp_file = catfile( $Bin,
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm]: I tried, but could not make it work with the streams and pipes and parenthesis. Any suggestions on how to do it?

**Comment by henrikstranneheim** [comment]
It is called in lib/MIP/Check/Pipeline.pm, but only for rd_dna pipe as of yet. Not sure it would work for RNA, and for a rerun I guess the gender should have been updated upstreams. However, we could add it to the DRAGEN pipe.

File: lib/MIP/Parse/Gender.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (jemten) [reply lib/MIP/Parse/Gender.pm]: Nothing more clever then giving the array ref directly to IPC::Open, but I'm guessing that you've already tried this. I say go ahead and merge and then we can take a look at it later if we feel the need :)

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm]: Yeah, I did - a bit annoying to not be able to just do it directly. On the other hand this works just fine and the bash shell does what is does best in this case. I will merge.

**Pull request merged**